src/plugins/data_view_management/public/components/state_helpers.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { map, distinctUntilChanged, Observable } from 'rxjs';
+import useObservable from 'react-use/lib/useObservable';
+import { isEqual } from 'lodash';
+
+// Use an observable for react state WITH a selector function
+export const useObservableSelector = <S, R>(
+  observable: Observable<S>,
+  selector: (arg0: S) => R,
+  defaultValue: R
+): R => useObservable(observable.pipe(map(selector), distinctUntilChanged(isEqual)), defaultValue);
```

Comment:
I'm hesitant to do this because it would be better to avoid thinking about such lifecycle concerns if possible. `useObservable` is really just a wrapper around `useState`. Perhaps `useObservable` could be modified to take accept a function that returns an observable.

Reply:
Vadim says my PR could cause changes.

Is there reason to believe that creating new observables is somehow expensive?

Reply:
I moved individual value observable creation into the service as it simplifies lifecycle concerned. I see the service as being inherently tied to the UI.